server_handlers.go

```diff
@@ -119,9 +127,22 @@ func serverHandshakeHandler(c *Conn) (*alert, error) {
 				}
 			}
 
-			c.state.masterSecret, err = prfMasterSecret(preMasterSecret, clientRandom, serverRandom, c.state.cipherSuite.hashFunc())
-			if err != nil {
-				return &alert{alertLevelFatal, alertInternalError}, err
+			if c.state.extendedMasterSecret {
+				var sessionHash []byte
+				sessionHash, err = c.handshakeCache.sessionHash(c.state.cipherSuite.hashFunc())
```

Comment:
`sessionHash, err := c.handshakeCache.sessionHash(c.state.cipherSuite.hashFunc())`

Reply:
Fixed!

Reply:
Actually lint fails with it this way:

```
client_handlers.go:19:16: declaration of "err" shadows declaration at client_handlers.go:9 (govet)
		sessionHash, err := c.handshakeCache.sessionHash(c.state.cipherSuite.hashFunc())
		             ^
server_handlers.go:131:18: declaration of "err" shadows declaration at server_handlers.go:106 (govet)
				sessionHash, err := c.handshakeCache.sessionHash(c.state.cipherSuite.hashFunc())
```